src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedContainer.cs

```diff
@@ -123,16 +125,40 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
  
         public override ImmutableArray<Symbol> GetMembers()
         {
+            var builder = ArrayBuilder<Symbol>.GetInstance();
+            if (this.TypeKind == TypeKind.Struct)
+            {
+                // Add an extra object field for EnC.
+                builder.Add(GetStructFieldCaptureObject());
+            }
+
             Symbol constructor = this.Constructor;
-            return (object)constructor == null ? ImmutableArray<Symbol>.Empty : ImmutableArray.Create(constructor);
+            if ((object)constructor != null)
+            {
+                builder.Add(constructor);
+            }
+
+            return builder.ToImmutableAndFree();
         }
 
         public override ImmutableArray<Symbol> GetMembers(string name)
         {
```

Comment:
Seems like this method should just call `GetMembers()` and filter. I don't know why it's doing what it's doing right now, but it seems weird.